test/FunctionsV2/CorrelationEndToEndTests.cs

```diff
@@ -191,134 +196,88 @@ internal async Task<Tuple<List<OperationTelemetry>, List<ExceptionTelemetry>>>
             };
             DurableTaskOptions options = new DurableTaskOptions();
             options.Tracing = traceOptions;
-            var sendAction = new Action<ITelemetry>(
-                delegate(ITelemetry telemetry) { sendItems.Enqueue(telemetry); });
 
             string siteNameEnvironmentVarName = "WEBSITE_SITE_NAME";
             string siteNameEnvironmentVarValue = TestSiteName;
+
             var mockNameResolver = GetNameResolverMock(new[] { (siteNameEnvironmentVarName, siteNameEnvironmentVarValue) });
 
+            OptionsWrapper<DurableTaskOptions> optionsWrapper = new OptionsWrapper<DurableTaskOptions>(options);
+
+            TelemetryConfiguration telemetryConfiguration = new TelemetryConfiguration();
+            string connStringValue = "InstrumentationKey=xxxx;IngestionEndpoint =https://xxxx.applicationinsights.azure.com/;LiveEndpoint=https://xxxx.livediagnostics.monitor.azure.com/";
+            telemetryConfiguration.ConnectionString = connStringValue;
+
+            // Set up the custom telemetry channel
+            TestTelemetryChannel testTelemetryChannel = new TestTelemetryChannel();
+            telemetryConfiguration.TelemetryChannel = testTelemetryChannel;
+
+            TelemetryActivator activator = new TelemetryActivator(optionsWrapper, mockNameResolver.Object);
```

Comment:
This might have impact on other tests if you don't dispose it.

```suggestion
            using TelemetryActivator activator = new TelemetryActivator(optionsWrapper, mockNameResolver.Object);
```